### PR TITLE
Add create_column_families to C api

### PR DESCRIPTION
**Pull request by schmidek**

### CODE REVIEW

File: db/c.cc [hunk 1]
```diff
@@ -986,6 +986,33 @@ rocksdb_column_family_handle_t* rocksdb_create_column_family(
...
+  std::vector<ColumnFamilyHandle*> handles;
+  std::vector<std::string> names;
+  for (int i = 0; i != num_column_families; ++i) {
+    names.push_back(std::string(column_family_names[i]));
+  }
+  SaveError(errptr,
+      db->rep->CreateColumnFamilies(ColumnFamilyOptions(column_family_options->rep),
+        names, &handles));
+
+  rocksdb_column_family_handle_t** c_handles = static_cast<rocksdb_column_family_handle_t**>(malloc(sizeof(rocksdb_column_family_handle_t*) * num_column_families));
```
Comment (siying) [inline db/c.cc:1003]: In this code base, a line is 80-char maximum. Please wrap those lines that are longer.

**Review by siying — COMMENTED** [review]
Thank you for your contribution. Can you add a test in c_test to cover the new code?

File: db/c.cc (hunk 1 above)
Comment (schmidek) [reply db/c.cc]: Fixed

**Comment by schmidek** [comment]
Added a test for the new code in c_test

File: db/c.cc [hunk 2]
```diff
@@ -1057,6 +1057,35 @@ rocksdb_column_family_handle_t* rocksdb_create_column_family(
...
+    names.push_back(std::string(column_family_names[i]));
+  }
+  SaveError(errptr, db->rep->CreateColumnFamilies(
+                        ColumnFamilyOptions(column_family_options->rep), names,
+                        &handles));
+
+  rocksdb_column_family_handle_t** c_handles =
+      static_cast<rocksdb_column_family_handle_t**>(malloc(
+          sizeof(rocksdb_column_family_handle_t*) * num_column_families));
+  for (int i = 0; i != num_column_families; ++i) {
```
Comment (ajkr) [inline db/c.cc:1076]: `handles` is not guaranteed to have size `num_column_families` on error: https://github.com/facebook/rocksdb/blob/ca50ccc71a1ce89008b4737e74f321b8df8a3b5b/include/rocksdb/db.h#L357-L359

Should we iterate up to `handles.size()` instead? Although, that introduces a problem that the `rocksdb_create_column_families()` caller does not know the number of column families that were created. Perhaps you can add a `size_t*` argument to mitigate that, similar to `rocksdb_list_column_families()`.

**Review by ajkr — CHANGES_REQUESTED** [review]
LGTM besides the error case

File: db/c.cc (hunk 2 above)
Comment (schmidek) [reply db/c.cc]: Thanks, I had missed that detail.

**Review by ajkr — APPROVED** [review]
LGTM, thanks!

**Pull request closed without merging**